arch/lkl/kernel/syscalls.c

```diff
@@ -142,11 +170,23 @@ int syscalls_init(void)
 			return -1;
 	}
 
+	if (kernel_thread(idle_host_task_loop, NULL, CLONE_FLAGS) < 0) {
+		if (lkl_ops->tls_free)
+			lkl_ops->tls_free(task_key);
+		return -1;
+	}
+
 	return 0;
 }
 
 void syscalls_cleanup(void)
 {
+	struct thread_info *ti = task_thread_info(idle_host_task);
```

Comment:
I think there is chance that this idle_host_task never runs. Safer to check if it's null first?